Scripts/Player.cs

```diff
@@ -1,142 +1,221 @@
-            cursorPointOnFloor.y = GlobalTransform.origin.y;
-
-            return cursorPointOnFloor;
-        }
-        return null;
-    }
-}
+using System;
+using Godot;
+using Godot.Collections;
+
+public class Player : KinematicBody
+{
+    [Export]
+    protected float MaxSpeed = 25f;
+
+    [Export]
+    protected float TurningResistance = 0.2f;
+
+    [Export]
+    protected float MaxAccelerationFactor = 0.3f;
+
+    [Export]
+    protected float MaxDecelerationFactor = 0.1f;
+
+    [Export]
+    protected float WalkingSlowFactor = 0.7f;
+    protected float RotationSpeed = 30f;
+
+    private float WalkingSpeed
+    {
+        get
+        {
+            return WalkingSlowFactor * MaxSpeed;
+        }
+    }
+
+    private float MaxAcceleration
+    {
+        get
+        {
+            return MaxAccelerationFactor * MaxSpeed;
+        }
+    }
+    private float MaxDeceleration
+    {
+        get
+        {
+            return MaxDecelerationFactor * MaxSpeed;
+        }
+    }
+
+    private bool IsSprinting
+    {
+        get
+        {
+            return Input.IsActionPressed("movement_sprint");
+        }
+    }
+
+    private Vector2 mousePosition;
+    private Vector3 velocity;
+    private Camera camera;
+    private InputMode inputMode = InputMode.Keyboard;
+    private Vector3 previousFaceDirection = Vector3.Forward;
+    private Weapon weapon;
+
+    public override void _Ready()
+    {
+        camera = GetParent().GetNode<Camera>("Camera");
+        weapon = GetNode<Weapon>("Weapon");
+    }
+
+    public override void _Input(InputEvent @event)
+    {
+        // Mouse in viewport coordinates
+        if (@event is InputEventMouseMotion eventMouseMotion)
+        {
+            mousePosition = eventMouseMotion.Position;
+            inputMode = InputMode.Keyboard;
+        }
+    }
+
+    public override void _PhysicsProcess(float delta)
+    {
+        Vector2 weightedDirection = GetWeightedMovementDirection();
+        Move(weightedDirection, delta);
+        float angle = GetFaceDirection().AngleTo(previousFaceDirection);
+        angle = Math.Min(angle, RotationSpeed * delta);
+        Vector3 newFaceDirection = previousFaceDirection.Rotated(GetFaceDirection().Cross(previousFaceDirection), -angle);
+        LookAt(Translation + newFaceDirection, Vector3.Up);
+        previousFaceDirection = newFaceDirection.Normalized();
+
+        // So that the global rotation of the weapon will be zero
+        weapon.Rotation = -Rotation;
+        weapon.AttackDirection = GetAttackDirection();
+    }
+
+    public void Move(Vector2 weightedDirection, float delta)
+    {
+        float targetSpeed = IsSprinting ? MaxSpeed : WalkingSpeed;
+        Vector3 targetVelocity = targetSpeed * new Vector3(weightedDirection.x, 0, weightedDirection.y);
+
+        Vector3 targetAcceleration = targetVelocity - velocity;
+
+
+        Vector3 actualAcceleration = targetAcceleration;
```

Comment:
2 newlines?

Reply:
Oh yup I removed this in subsequent commits